LEGO1/lego/legoomni/src/actors/skateboard.cpp

```diff
@@ -17,17 +20,44 @@ SkateBoard::SkateBoard()
 	NotificationManager()->Register(this);
 }
 
-// STUB: LEGO1 0x10010000
+// FUNCTION: LEGO1 0x1000ff80
+SkateBoard::~SkateBoard()
+{
+	ControlManager()->Unregister(this);
+	NotificationManager()->Unregister(this);
+}
+
+// FUNCTION: LEGO1 0x10010000
 MxResult SkateBoard::Create(MxDSAction& p_dsAction)
 {
-	// TODO
-	return SUCCESS;
+	MxResult result = IslePathActor::Create(p_dsAction);
+	if (result == SUCCESS) {
+		this->m_world = CurrentWorld();
+		this->m_world->Add(this);
+		// The type `Pizza` is an educated guesss, inferred from VTable0xe4() below
+		Pizza* findResult = (Pizza*) CurrentWorld()->Find(*g_isleScript, 0x49d);
```

Comment:
Please use the action constants found in the script headers instead of magic values. In this case you will find it in `LEGO1\lego\legoomni\include\actions\isle_actions.h` (`c_Pizza_Actor`)

Reply:
I was wondering if there is a more type-safe way to declare `CurrentWorld()->Find()`. I have tried a bit and have already found other violations of this rule. Will open another PR if I get somewhere.